src/cpp/dds/topic/Topic.cpp

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2020, Proyectos y Sistemas de Mantenimiento SL (eProsima).
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+/**
+ * @file
+ */
+
+/*
+ * OMG PSM class declaration
+ */
+#include <dds/topic/Topic.hpp>
+#include <dds/topic/detail/Topic.hpp>
+#include <dds/topic/TopicListener.hpp>
+
+namespace dds {
+namespace topic {
+
+//Topic::Topic(
+//        const dds::domain::DomainParticipant& dp)
+//    : ::dds::core::Reference<detail::Topic>(
+//        new detail::Topic(
+//            dp.delegate().get(),
+//            dp.default_Topic_qos(),
+//            nullptr,
+//            dds::core::status::StatusMask::all()))
+//    , participant_(nullptr)
+//{
+//}
```

Comment:
Uncomment also this function

Reply:
I had to add `topic_name` and `type_name` parameters for this to work.